packages/contracts-periphery/contracts/foundry-tests/Optimist.t.sol

```diff
@@ -0,0 +1,371 @@
+
+        vm.label(alice_admin, "alice_admin");
+        vm.label(bob, "bob");
+        vm.label(sally, "sally");
+        _initializeContracts();
+    }
+
+    function _initializeContracts() internal {
+        attestationStation = new AttestationStation();
+        optimist = new Optimist(name, symbol, alice_admin, attestationStation);
+    }
+}
+
+contract OptimistTest is Optimist_Initializer {
+    function setUp() public {
+        super._setUp();
+        _initializeContracts();
+    }
+
+    function test_optimist_initialize() external {
+        // expect name to be set
+        assertEq(optimist.name(), name);
+        // expect symbol to be set
+        assertEq(optimist.symbol(), symbol);
+        // expect attestationStation to be set
+        assertEq(address(optimist.ATTESTATION_STATION()), address(attestationStation));
+    }
+
+    /**
+     * @dev Bob should be able to mint an NFT if he is whitelisted
+     * by the attestation station and has a balance of 0
+     */
+    function test_optimist_mint_happy_path() external {
+        // bob should start with 0 balance
+        assertEq(optimist.balanceOf(bob), 0);
+
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint an NFT
+        vm.prank(bob);
+        optimist.mint(bob);
+        // expect the NFT to be owned by bob
+        assertEq(optimist.ownerOf(256), bob);
+        assertEq(optimist.balanceOf(bob), 1);
+    }
+
+    /**
+     * @dev Sally should be able to mint a token on behalf of bob
+     */
+    function test_optimist_mint_secondary_minter() external {
+        // whitelist bob
+        AttestationStation.AttestationData[]
+            memory attestationData = new AttestationStation.AttestationData[](1);
+        // we are using true but it can be any non null value
+        attestationData[0] = AttestationStation.AttestationData({
+            about: bob,
+            key: bytes32("optimist.can-mint"),
+            val: bytes("true")
+        });
+        vm.prank(alice_admin);
+        attestationStation.attest(attestationData);
+
+        // mint as sally instead of bob
+        vm.prank(sally);
+        optimist.mint(bob);
+
+        // expect the NFT to be owned by bob
+        assertEq(optimist.ownerOf(256), bob);
+        assertEq(optimist.balanceOf(bob), 1);
+    }
+
+    /**
+     * @dev Bob should not be able to mint an NFT if he is not whitelisted
+     */
+    function test_optimist_mint_no_attestation() external {
+        vm.prank(bob);
+        vm.expectRevert("Optimist: address is not whitelisted");
```

Comment:
not opposed to allowlisted here instead

Reply:
plz can we change whitelist, I still have PTSD from our last whitelist

Reply:
and error message can be something like "cannot mint yet"

Reply:
We aren't using this in the web app so I safely made this change everywhere in the contract